asciidoc/workloads/azure/backend/java/setting_up_cosmos_db_locally_java.adoc

```diff
@@ -0,0 +1,117 @@
+= Cosmos DB Emulator
+:description: How to set up the Cosmos DB Emulator
+:imagesdir: ../../../../../static/img
+:keywords: cosmos db, emulator , settings, template, certificate
+
+== Set up Cosmos DB Emulator locally
+
+The Azure CosmosDB Emulator provides a local environment that emulates the Azure Cosmos DB
+service for development purposes. Using the Azure CosmosDB Emulator, you can develop and test
+your application locally, without creating an Azure subscription or incurring any costs.
+Latest instruction can be found in https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator?tabs=cli%2Cssl-netstd21[here]
+
+== Using the Cosmos DB Emulator to run the database locally
+
+Download and install the latest version of **CosmosDB Emulator**. Once it's installed, it will be
+available on https://localhost:8081/_explorer/index.html[localhost].
+
+=== Windows
+You can also find the Emulator from Start Menu and type Azure CosmosDB Emulator.
+
+=== Unix
+
+The **CosmosDB Emulator** service is primarily a Windows service, so using it in Unix/Linux/MacOS requires a few
+additional steps. One of the simplest ways of running the emulator is to download
+the https://docs.microsoft.com/en-us/azure/cosmos-db/linux-emulator[Linux Emulator] for Docker.
+Please follow the installation instructions, with one exception: when running the Docker image, don't set
+the `AZURE_COSMOS_EMULATOR_IP_ADDRESS_OVERRIDE=$ipaddr`. Instead, set it to `127.0.0.1`.
+
+
+The value for `COSMOSDB_KEY` can be found within the emulator. See Primary Key:
+
+image::cosmosdb_emulator_3.png[cosmosdb]
+
+Please, take a note of the cosmosdb URI, databaseName and key, as they will be needed to configure the
+application in a further step.
+
+== Create the Cosmos DB structure
+
+Based on the template the CosmosDB has to contain a fixed structure.
+
+Create a collection `Stacks` with a container id `menu` and the partition key `/id`.
+
+Create a database called Ensono Stacks for the application, and a database called `CosmosDBPackage` for the
+integration tests of the CosmosDB package (using the same container id and partition key as the main database).
+
+image::cosmosdb_emulator_1.png[cosmosdb]
+
+The **Ensono Stacks** database should have a container called **Menu** partitioned by **/id**.
+
+image::cosmosdb_emulator_2.png[cosmosdb]
+
+== Export Azure Cosmos DB Emulator certificates to use in the Java Application
+
+=== Windows
```

Comment:
You could consider putting the longer sections like this in collapsible blocks, so its a bit easier to see what is windows and what is unix. I've done this for one or two other when its not just a one line command.